### PR TITLE
rf reg cookbook

### PR DESCRIPTION
**Pull request by Saurabh7**

### CODE REVIEW

**Comment by Saurabh7** [comment]
@karlnapf I will send data,
but I think its good to set num of threads here in some way, currently 

```
rand_forest.parallel.set_num_threads(1)
rand_forest.train(features_train)
```
gives

```
/shogun/build/examples/meta/cpp/regression/random_forest_regression.cpp:16:1: error: ‘rand_forest’ was not declared in this scope
 rand_forest->train(features_train);

```

**Comment by karlnapf** [comment]
Ah yeah, I think we need some SWIG compatible wrapper method to set the number of threads. Like for init seed.
This thing you do is not compatible with the meta grammar.
I assume get_global_parallel doesnt work?

File: doc/cookbook/source/examples/regression/random_forest_regression.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
+=============
+Random Forest
+=============
+
+A Random Forest is an ensemble learning method which implements multiple decision trees during training. It predicts by using a combination rule on the outputs of individual decision trees.
+
+See :cite:`Breiman2001` for a detailed introduction.
```
Comment (karlnapf) [inline doc/cookbook/source/examples/regression/random_forest_regression.rst]: pls reference the classification cookbook

File: examples/meta/src/regression/random_forest_regression.sg [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+RegressionLabels labels_train(f_labels_train)
+RegressionLabels labels_test(f_labels_test)
+#![create_features]
+
+#![create_combination_rule]
+MeanRule mean_rule()
+#![create_combination_rule]
+
+#![create_instance]
+RandomForest rand_forest(1,10)
```
Comment (karlnapf) [inline examples/meta/src/regression/random_forest_regression.sg]: can you make 1 and 10 explicit variables and mention them?

**Comment by karlnapf** [comment]
Pretty good, this is mergable once the minor things are addressed,
we can hold the integration data back until we sorted out the thread setting...

**Comment by Saurabh7** [comment]
updated. 
ah something else :hand:  `:doc:random_forest` doesnt link properly here just saw http://buildbot.shogun-toolbox.org/cookbook_pr/208f4b16af90cd66ef38fe5c434f3d781ad147e8/examples/regression/random_forest_regression.html.

**Pull request merged**